llama_stack/providers/inline/vector_io/sqlite_vec/sqlite_vec.py

```diff
@@ -184,17 +226,66 @@ def _execute_query():
                 connection.close()
 
         rows = await asyncio.to_thread(_execute_query)
+        chunks, scores = [], []
+        for row in rows:
+            _id, chunk_json, distance = row
+            score = 1.0 / distance if distance != 0 else float("inf")
+            if score < score_threshold:
+                continue
+            try:
+                chunk = Chunk.model_validate_json(chunk_json)
+            except Exception as e:
+                logger.error(f"Error parsing chunk JSON for id {_id}: {e}")
+                continue
+            chunks.append(chunk)
+            scores.append(score)
+        return QueryChunksResponse(chunks=chunks, scores=scores)
+
+    async def query_keyword(
+        self,
+        query_string: str | None,
```

Comment:
```suggestion
        query_string: Optional[str],
```

Reply:
This would be a required parameter to allow keyword search.